app/Http/Controllers/UserAdminController.php

```diff
@@ -128,8 +128,13 @@ public function addMembership($id, Request $request)
             return Redirect::back();
         }
 
-        $member = Member::create();
-        $member->user()->associate($user);
+        if (!$user->member) {
+            $member = Member::create();
+            $member->user()->associate($user);
+        } else {
+            $member = $user->member;
```

Comment:
In this case, I'd re-set the date of the start of membership to the current time, since this is when the membership actually starts.